stdlib/experimental/prometheus/scrape.go

```diff
@@ -0,0 +1,334 @@
+
+	// Parse the entire body into a slice of Metrics
+	err = p.parse(resp.Body, resp.Header)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *PrometheusIterator) parse(r io.Reader, header http.Header) (err error) {
+	var parser expfmt.TextParser
+	now := time.Now()
+
+	mediatype, params, err := mime.ParseMediaType(header.Get("Content-Type"))
+	if err != nil {
+		return err
+	}
+
+	// Prepare output
+	metricFamilies := make(map[string]*dto.MetricFamily)
+	if mediatype == "application/vnd.google.protobuf" &&
+		params["encoding"] == "delimited" &&
+		params["proto"] == "io.prometheus.client.MetricFamily" {
+		for {
+			mf := &dto.MetricFamily{}
+			if _, err := pbutil.ReadDelimited(r, mf); err != nil {
+				if err == io.EOF {
+					break
+				}
+				return fmt.Errorf("reading metric family protocol buffer failed: %s", err)
+			}
+			metricFamilies[mf.GetName()] = mf
+		}
+	} else {
+		metricFamilies, err = parser.TextToMetricFamilies(r)
+		if err != nil {
+			return fmt.Errorf("reading text format failed: %s", err)
+		}
+	}
+
+	p.metrics = make([]Metric, 0)
+
+	// read metrics
+	for name, family := range metricFamilies {
+		for _, m := range family.Metric {
+			// read tags
+			tags := makeLabels(m)
+
+			// read fields
+			fields := make(map[string]interface{})
+			fields = getNameAndValue(m)
+
+			// Store data needed to build flux Table
+			if len(fields) > 0 {
+				tm := now
+				if m.TimestampMs != nil && *m.TimestampMs > 0 {
+					tm = time.Unix(0, *m.TimestampMs*1000000)
+				}
+				met := Metric{
+					Timestamp: tm,
+					Tags:      tags,
+					Fields:    fields,
+					Name:      name,
+				}
+				p.metrics = append(p.metrics, met)
+			}
+		}
+	}
+	return nil
+}
+
+func (p *PrometheusIterator) Fetch(ctx context.Context) (bool, error) {
+	// Iterate over all Metrics in the slice
+	p.i++
+	if p.i < len(p.metrics) {
+		p.met = p.metrics[p.i]
+		return true, nil
+	}
+
+	// No more metrics to return
+	return false, nil
+}
+
+func (p *PrometheusIterator) Decode(ctx context.Context) (table flux.Table, err error) {
+
+	// Unpacking Fields map
+	var val interface{}
+	for _, v := range p.met.Fields {
+		val = v
+	}
+
+	groupKey := execute.NewGroupKeyBuilder(nil)
+	groupKey.AddKeyValue("_measurement", values.New("prometheus"))
+	groupKey.AddKeyValue("_field", values.New(p.met.Name))
+
+	gk, err := groupKey.Build()
+	if err != nil {
+		return nil, err
+	}
+	builder := execute.NewColListTableBuilder(gk, p.administration.Allocator())
+
+	builder.AddCol(flux.ColMeta{
+		Label: "_time",
+		Type:  flux.TString,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "_value",
+		Type:  flux.TFloat,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "_measurement", // data source
+		Type:  flux.TString,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "_field", // promethus metric name
+		Type:  flux.TString,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "url",
+		Type:  flux.TString,
+	})
+
+	builder.AppendValue(0, values.New(p.met.Timestamp.String()))
```

Comment:
You are appending a string to a timestamp column here.  Unless I'm mistaken I think you want to use `values.ConvertTime` instead?